src/uu/df/src/table.rs

```diff
@@ -261,84 +259,163 @@ impl<'a> DisplayRow<'a> {
+            };
+
+            headers.push(header);
+        }
+
+        headers
+    }
 }
 
-impl<'a> Header<'a> {
-    /// Instantiate this struct.
-    pub(crate) fn new(options: &'a Options) -> Self {
-        Self { options }
+/// The output table.
+pub(crate) struct Table {
+    alignments: Vec<Alignment>,
+    rows: Vec<Vec<String>>,
+    widths: Vec<usize>,
+}
+
+impl Table {
+    pub(crate) fn new(options: &Options, filesystems: Vec<Filesystem>) -> Self {
+        let headers = Header::get_headers(options);
+        let mut widths: Vec<_> = options
+            .columns
+            .iter()
+            .enumerate()
+            .map(|(i, col)| Column::min_width(col).max(headers[i].len()))
+            .collect();
+
+        let mut rows = vec![headers];
+
+        // The running total of filesystem sizes and usage.
+        //
+        // This accumulator is computed in case we need to display the
+        // total counts in the last row of the table.
+        let mut total = Row::new("total");
+
+        for filesystem in filesystems {
+            // If the filesystem is not empty, or if the options require
+            // showing all filesystems, then print the data as a row in
+            // the output table.
+            if options.show_all_fs || options.show_listed_fs || filesystem.usage.blocks > 0 {
+                let row = Row::from(filesystem);
+                let fmt = RowFormatter::new(&row, options);
+                let values = fmt.get_values();
+                total += row;
+
+                for (i, value) in values.iter().enumerate() {
+                    if value.len() > widths[i] {
+                        widths[i] = value.len();
+                    }
+                }
+
+                rows.push(values);
+            }
+        }
+
+        if options.show_total {
+            let total_row = RowFormatter::new(&total, options);
+            rows.push(total_row.get_values());
+        }
+
+        Self {
+            rows,
+            widths,
+            alignments: Self::get_alignments(&options.columns),
+        }
+    }
+
+    fn get_alignments(columns: &Vec<Column>) -> Vec<Alignment> {
+        let mut alignments = Vec::new();
+
+        for column in columns {
+            alignments.push(Column::align(column));
+        }
+
+        alignments
     }
 }
 
-impl fmt::Display for Header<'_> {
+impl fmt::Display for Table {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        for column in &self.options.columns {
-            match column {
-                Column::Source => write!(f, "{0: <16} ", "Filesystem")?,
-                // `Display` is implemented for `BlockSize`, but
-                // `Display` only works when formatting an object into
-                // an empty format, `{}`. So we use `format!()` first
-                // to create the string, then use `write!()` to align
-                // the string and pad with spaces.
```

Comment:
Is this workaround no longer necessary?

Reply:
No, I simply use `to_string()`, and the padding is no longer done in `Header` but in `Table`.